src/eko/quad_ker_base.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+
+import numba as nb
+import numpy as np
+
+from . import interpolation, mellin
+
+spec = [
+    ("ker_mode", nb.int8),
+    ("u", nb.float64),
+    ("is_log", nb.boolean),
+    ("logx", nb.float64),
+    ("areas", nb.float64[:, :]),
+    ("mode0", nb.int64),
+    ("mode1", nb.int64),
+]
+
+MODE_EVOLUTION = 0
+"""Evolution mode to combine anomalous dimensions"""
+
+MODE_MATCHING = 1
+"""Matching mode to combine operator matching elements"""
+
+
+@nb.experimental.jitclass(spec)
+class QuadKerBase:
+    """Manage the common part of Mellin inversion integral.
+
+    Parameters
+    ----------
+    ker_mode : int
+        kernel mode: 0 for anomalous dimension, 1 for matching
+    u : float
+        quad argument
+    is_log : boolean
+        is a logarithmic interpolation
+    logx : float
+        Mellin inversion point
+    areas : tuple
+        basis function configuration
+    mode0 : str
+        first sector element
+    mode1 : str
+        second sector element
+    """
+
+    def __init__(self, ker_mode, u, is_log, logx, areas, mode0, mode1):
+        self.ker_mode = ker_mode
+        self.u = u
+        self.is_log = is_log
+        self.logx = logx
+        self.areas = areas
+        self.mode0 = mode0
+        self.mode1 = mode1
+
+    @property
+    def path(self):
+        """Returns the associated instance of :class:`eko.mellin.Path`"""
+        return mellin.Path(self.u, self.logx, self.is_singlet)
+
+    @property
+    def n(self):
+        """Returns the Mellin moment N"""
+        return self.path.n
+
+    @property
+    def is_singlet(self):
+        """Returns the Mellin moment N"""
+        return self.mode0 in [100, 21, 90]
+
+    def select_element(self, ker):
+        """Select correct element of (matrix-valued) kernel.
+
+        Parameters
+        ----------
+        ker : numpy.complex or numpy.ndarray
+            integration kernel
+        mode0 : int
+            id for first sector element
+        mode1 : int
+            id for second sector element
+
+        Returns
+        -------
+        ker : complex
+            singlet integration kernel element
+        """
+        # OME mode
+        if self.ker_mode == MODE_MATCHING:
+            if self.is_singlet:
+                indices = {21: 0, 100: 1, 90: 2}
+            else:
+                indices = {200: 0, 91: 1}
+            return ker[indices[self.mode0], indices[self.mode1]]
+        # AD mode
+        k = 0 if self.mode0 == 100 else 1
+        l = 0 if self.mode1 == 100 else 1
```

Comment:
I had a look at code factor, and most of the remaining issues are with poor names.

On one side, the immediate feeling is that it is too picky, and for sure it is not able to recognize long living variables from local indices and so on. But on the other, it is only (or mostly) complaining about one letter variables, that we might really consider too little, since it is for sure not meaningful at all (almost anonymous).

Here, e.g., I would make the minimal effort to improve, and just call the two of them `fl1` and `fl2` (or `fli` and `flo`, for I/O).